editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/HomepagePlanningForm.tsx

```diff
@@ -0,0 +1,123 @@
+import RadioGroup from "@mui/material/RadioGroup";
+import BasicRadio from "@planx/components/shared/Radio/BasicRadio";
+import { useFormik } from "formik";
+import React, { ChangeEvent, useState } from "react";
+import InputDescription from "ui/editor/InputDescription";
+import Input from "ui/shared/Input";
+import InputRow from "ui/shared/InputRow";
+import InputRowLabel from "ui/shared/InputRowLabel";
+
+import { SettingsForm } from "../shared/SettingsForm";
+import { FormProps } from ".";
+export default function HomepagePlanningForm({
+  formikConfig,
+  onSuccess,
+}: FormProps) {
+  const [showPlanningInputs, setShowPlanningInputs] = useState(false);
+
+  const formik = useFormik({
+    ...formikConfig,
+    onSubmit(values, { resetForm }) {
+      onSuccess();
+      resetForm({ values });
+    },
+  });
+
+  const onChangeFn = (type: string, event: ChangeEvent<HTMLInputElement>) =>
+    formik.setFieldValue(type, event.target.value);
+
+  const boolTransform = (event: ChangeEvent<HTMLInputElement>) => {
+    if (event.target.value === "yes") {
+      setShowPlanningInputs(true);
+      return true;
+    } else if (event.target.value === "no") {
+      setShowPlanningInputs(false);
+      return false;
+    }
+  };
```

Comment:
Have a look at https://github.com/theopensystemslab/planx-new/pull/3287

@ianjon3s has recently made some changes in-Editor to use switches to control boolean fields instead of checkboxes or radio buttons. Using this will give us a more consistent UI, as well has handling some of these awkward castings of strings → booleans.

Reply:
Yeah I had this as a question for Ian on Figma, I wasn't sure the situations I should apply it to (turning on / off a setting or feature compared with answering a question). Happy to switch it to a ``Switch``

Reply:
Resolving since the use of a Switch is no longer relevant